manifests/params.pp

```diff
@@ -184,6 +183,12 @@
   $server_http_port           = 8139
   $server_http_allow          = []
 
+  # use puppetserver (JVM) or puppet master (Ruby)?
+  $server_implementation = $aio_package ? {
+    true    => 'puppetserver',
+    default => 'master',
+  }
+
```

Comment:
While I don't have strong objections against the globals pattern, I'd prefer to avoid it if possible. In this case I agree with @mmoll that you can still override ```$server_implementation``` and that's the only place it's used.